src/explorer/loadedScriptsExplorer.ts

```diff
@@ -0,0 +1,268 @@
+        return this.createIfNeeded(session.id, () => new SessionTreeItem(session));
+    }
+}
+
+const URL_REGEXP = /^(https?:\/\/[^/]+)(\/.*)$/;
+
+class SessionTreeItem extends BaseTreeItem {
+
+    private _session: vscode.DebugSession;
+    private _initialized: boolean;
+
+    constructor(session: vscode.DebugSession) {
+        super(session.name, vscode.TreeItemCollapsibleState.Expanded);
+        this._initialized = false;
+        this._session = session;
+    }
+
+    getChildren(): ProviderResult<BaseTreeItem[]> {
+
+        if (!this._initialized) {
+            this._initialized = true;
+            //return listLoadedScripts(this._session).then(paths => {
+            return Promise.resolve([]).then(paths => {
+                if (paths) {
+                    paths.forEach(path => this.addPath(path));
+                }
+                return super.getChildren();
+            });
+        }
+
+        return super.getChildren();
+    }
+
+    protected compare(a: BaseTreeItem, b: BaseTreeItem): number {
+        const acat = this.category(a);
+        const bcat = this.category(b);
+        if (acat !== bcat) {
+            return acat - bcat;
+        }
+        return super.compare(a, b);
+    }
+
+    /**
+     * Return an ordinal number for folders
+     */
+    private category(item: BaseTreeItem): number {
+
+        // workspace scripts come at the beginning in "folder" order
+        if (item instanceof FolderTreeItem) {
+            return item.folder.index;
+        }
+
+        // <...> come at the very end
+        if (/^<.+>$/.test(item.label)) {
+            return 1000;
+        }
+
+        // everything else in between
+        return 999;
+    }
+
+    addPath(source: Source): void {
+
+        let folder: vscode.WorkspaceFolder | undefined;
+        let url: string;
+        let p: string;
+
+        let path = source.path;
+
+        const match = URL_REGEXP.exec(path);
+        if (match && match.length === 3) {
+            url = match[1];
+            p = decodeURI(match[2]);
+        } else {
+            folder = vscode.workspace.getWorkspaceFolder(vscode.Uri.file(path));
+            p = trim(path);
+        }
+
+        let x: BaseTreeItem = this;
+        p.split(/[\/\\]/).forEach((segment, i) => {
+            if (segment.length === 0) {	// macOS or unix path
+                segment = '/';
+            }
+            if (i === 0 && folder) {
+                x = x.createIfNeeded(folder.name, () => new FolderTreeItem(<vscode.WorkspaceFolder>folder));
+            } else if (i === 0 && url) {
+                x = x.createIfNeeded(url, () => new BaseTreeItem(url));
+            } else {
+                x = x.createIfNeeded(segment, () => new BaseTreeItem(segment));
+            }
+        });
+
+        x.collapsibleState = vscode.TreeItemCollapsibleState.None;
+        x.setSource(this._session, source);
+    }
+}
+
+class FolderTreeItem extends BaseTreeItem {
```

Comment:
Investigate when a folder is going to be used.

Reply:
Turned out `FolderTreeItem` is not being used. The intermediate folder are created as BaseTreeItem as well. 

Deleted altogether.